File: app/src/main/java/art/coded/wireframe/model/local/ElementDao.java
package art.coded.wireframe.model.local;

import androidx.lifecycle.LiveData;
import androidx.paging.DataSource;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import art.coded.wireframe.model.entity.Element;

@Dao public interface ElementDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE) void insert(Element element);
    @Delete void delete(Element element);
    @Query("SELECT * from element_table") LiveData<List<Element>> getAll();
    @Query("SELECT * from element_table LIMIT 1") Element[] getAny();
    @Query("DELETE from element_table") void deleteAll();

    @Query("SELECT * from element_table")
    DataSource.Factory<Integer, Element> pagingSource();
}

File: app/src/ktx/java/art/coded/wireframe/model/local/ElementDao.kt
package art.coded.wireframe.model.local

import androidx.lifecycle.LiveData
import androidx.paging.DataSource
import androidx.room.*
import art.coded.wireframe.model.entity.Element

@Dao
interface ElementDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(element: Element?)

    @Delete
    fun delete(element: Element?)

    @get:Query("SELECT * from element_table")
    val all: LiveData<List<Element?>?>?

    @get:Query("SELECT * from element_table LIMIT 1")
    val any: Array<Element?>

    @Query("DELETE from element_table")
    fun deleteAll()

    @Query("SELECT * from element_table")
    fun pagingSource(): DataSource.Factory<Int, Element>
}